openshift/jenkins/Jenkinsfile.promote-to-test

```diff
@@ -0,0 +1,143 @@
+    VANITY_URL="https://pims-test.pathfinder.gov.bc.ca/"
+
+    // The name of your application
+    APP_NAME = "pims"
+
+    // Deployment configuration
+    DEV_ENV = "${DEPLOYMENT_ENVIRONMENT_TAGS[0]}"
+    TEST_ENV = "${DEPLOYMENT_ENVIRONMENT_TAGS[1]}"
+    PROD_ENV = "${DEPLOYMENT_ENVIRONMENT_TAGS[2]}"
+
+    // The name of the project namespace(s).
+    NAME_SPACE = "jcxjin"
+
+    // enable pipeline verbose debug output if greater than 0
+    DEBUG_OUTPUT = "false"
+  }
+  stages {
+    stage("Initialize") {
+      agent any
+      steps {
+        script {
+          // load supporting functions from external script files
+          cli = load "./openshift/jenkins/common.groovy"
+          notify = load "./openshift/jenkins/notify.groovy"
+
+          // ensure required secrets and credentials are available in the CI environment
+          // [add more secrets here if needed...]
+          sh "oc extract secret/rocket-chat-secrets --to=${env.WORKSPACE} --confirm"
+
+          // initialize CI environment values
+          env.ROCKET_DEPLOY_WEBHOOK = readFile("rocket-deploy-webhook").trim()
+
+          if (DEBUG_OUTPUT.equalsIgnoreCase("true")) {
+            // Force OpenShift Plugin directives to be verbose
+            openshift.logLevel(1)
+
+            // Print all environment variables
+            echo 'DEBUG - All pipeline environment variables:'
+            echo sh(returnStdout: true, script: 'env')
+          }
+        }
+      }
+    }
+    stage("Manual Approval") {
+      agent any
+      options {
+        timeout(time: 1, unit: "HOURS")
+      }
+      input {
+        message "Deploy to Staging/Test?"
+        ok "Yes, go ahead"
+      }
+      steps {
+        echo "[ci/cd] Promotion from DEV to STAGING (Test) was approved. Proceeding..."
+      }
+    }
+    stage("Promote API To Staging") {
+      agent any
+      environment {
+        COMP_NAME = "api"
+        SOURCE = DEV_ENV
+        DESTINATION = TEST_ENV
+      }
+      steps {
+        script {
+          openshift.withCluster() {
+            openshift.withProject() {
+              def image_name = "${APP_NAME}-${COMP_NAME}"
+              echo "[ci/cd]  Tagging ${image_name}:${SOURCE} for deployment to ${DESTINATION.toUpperCase()} ..."
+              openshift.tag("${image_name}@${SOURCE}", "${image_name}:${DESTINATION}")
+            }
+
+            openshift.withProject("${NAME_SPACE}-${DESTINATION}") {
+                def dc = openshift.selector("dc", [ app: APP_NAME, component: COMP_NAME ])
+                echo "[ci/cd]  Found ${dc.count()} DeploymentConfigs: ${dc.names()}"
+                // Wait for the deployment to complete.
+                // This will wait until the desired replicas are all available
+                dc.rollout().status()
+            }
+          }
+        }
+        echo "[ci/cd]  ${COMP_NAME.toUpperCase()} deployment complete!"
+      }
+    }
+    stage("Promote WEB To Staging") {
```

Comment:
/nit weren't we calling this FRONTEND previously (instead of WEB)?

Reply:
See the other Jenkins file where I renamed it everywhere. I'm fine calling it either. We can discuss tomorrow. As long as it's not APP which is way too similar to API, besides APP can mean so many things